src_c/math.c

```diff
@@ -792,6 +796,75 @@ vector_copy(pgVector *self)
     return (PyObject *)ret;
 }
 
+static PyObject *
+vector_clamp_magnitude(pgVector *self, PyObject *args, PyObject *kwargs)
+{
+    Py_ssize_t i;
+    pgVector *ret;
+
+    ret = (pgVector *)pgVector_NEW(self->dim);
+    if (ret == NULL)
+        return NULL;
+
+    for (i = 0; i < self->dim; ++i)
+        ret->coords[i] = self->coords[i];
+
+    vector_clamp_magnitude_ip(ret, args, kwargs);
+
+    return (PyObject *)ret;
+}
+
+static PyObject *
+vector_clamp_magnitude_ip(pgVector *self, PyObject *args, PyObject *kwargs)
+{
+    Py_ssize_t i;
+    double min_length = 0; /* Default minimum. */
+    double max_length;
+    double old_length;
+    double fraction;
+    static char *keywords[] = {"max_length", "min_length", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "d|d", keywords,
+                                     &max_length, &min_length)) {
+        return NULL;
+    }
+
+    /* Get magnitude of Vector */
+    old_length = sqrt(_scalar_product(self->coords, self->coords, self->dim));
```

Comment:
Optimization: I'm not sure how fast the sqrt() method is today... but why not avoid it here and only apply it when needed?

E.g. 
```
old_length_sq = sqrt(_scalar_product(self->coords, self->coords, self->dim));
if(old_length_sq == 0)
  ...
...
if(old_length_sq > max_length * max_length)
  fraction = max_length /  sqrt(old_length_sq);
  ...
...
```